### PR TITLE
Use config.local.js if present for grunt serve:dist

### LINKED ISSUE #1795: grunt serve:dist doesn't use config.local.js
This prevents log in from working with our default config.js and `oc cluster up` for one.

Related to #1325

@jwforres @rhamilto FYI

### PR DESCRIPTION
**Pull request by spadgett**
Fixes #1795

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -157,14 +157,21 @@ module.exports = function (grunt) {
...
+              '^/styles/extensions\.css$ /' + contextRoot + '/extensions/extensions.css [R]',
+              '^/scripts/extensions\.js$ /' + contextRoot + '/extensions/extensions.js [R]',
+              '^/$ /' + contextRoot + '/ [R=302]',
+              '^/' + contextRoot + '(.*)$ $1',
+              '!^/(config.js|(bower_components|scripts|images|styles|views|extensions)(/.*)?)$ /index.html [L]'
+            ];
+
+            // If config.local.js exists, use that instead of config.js.
+            if (grunt.file.exists('app/config.local.js')) {
+              rewriteRules.unshift('^/config.js$ /config.local.js [L]');
```
Comment (jwforres) [inline Gruntfile.js]: is this how we are doing it for the dev server?  the only problem i see with this is that if you delete your config.local.js file you have to restart your grunt server

Comment (spadgett) [inline Gruntfile.js]: No, we copy the `config.local.js` to `.tmp` for `grunt serve`, but I don't think that works here.

https://github.com/openshift/origin-web-console/blob/master/Gruntfile.js#L552-L563

I suspect copying has the same problem when you delete the file.

Comment (jwforres) [inline Gruntfile.js]: Ok, not a big deal since we rarely create/delete them

**Comment by jwforres** [comment]
[merge]  going ahead and merging since this doesn't affect shipped code

**Pull request merged**